### PR TITLE
Fixed bugs in the processing of quadratic elements

### PR DESCRIPTION
**Pull request by wenqing**
as titled.

### CODE REVIEW

File: METIS2ogs/partition.cpp [hunk 1]
```diff
@@ -93,16 +90,34 @@ enum PartType {by_element, by_node};
 
 int main(int argc, char* argv[])
 {
-   ifstream infile;
-   fstream ofile;
-   stringstream ss;
+   std::vector<std::string> cmd_args;
```
Comment (TomFischer) [inline METIS2ogs/partition.cpp]: Why not use TCLAP?!

File: Mesh/Mesh.cpp [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #include <cmath>
 #include <cfloat>
 #include <limits>
-#include <set>
+#include <algorithm>
 
 #include <stdio.h> // for binary output
```
Comment (TomFischer) [inline Mesh/Mesh.cpp]: `#include <cstdio>`?

File: Mesh/Mesh.cpp [hunk 3]
```diff
@@ -618,7 +618,7 @@ void Mesh::ConstructSubDomain_by_Elements(const string fname, const int num_part
    }
    max_dom++;
    part_in.close();
-   remove(str.c_str());
+   remove(str.c_str()); // Remove the metis file.
```
Comment (TomFischer) [inline Mesh/Mesh.cpp]: Du you mean 'std::remove`?

File: Mesh/Mesh.cpp [hunk 4]
```diff
@@ -979,10 +976,14 @@ void Mesh::ConstructSubDomain_by_Nodes(const MeshPartConfig mpc)
    vector<Node*> sbd_nodes_hq;     // nodes of a partition for quadratic element
    std::vector<std::vector<std::set<ConnEdge> > > vec_neighbors(num_parts); //NW
 
+   // Should be removed for openmp 
+   for (std::size_t i=0; i<elem_vector.size(); i++)
```
Comment (TomFischer) [inline Mesh/Mesh.cpp]: You could use a [range-based for loop](http://en.cppreference.com/w/cpp/language/range-for).

**Comment by TomFischer** [comment]
Sometimes functions and class of the standard template library are qualified by `std::`, often they are not. I would suggest to make things consistent and always use `std::`.

File: METIS2ogs/partition.cpp (hunk 1 above)
Comment (wenqing) [inline METIS2ogs/partition.cpp]: This tool is for ogs5, and I would like not to include third-party source code.

File: Mesh/Mesh.cpp (hunk 2 above)
Comment (wenqing) [inline Mesh/Mesh.cpp]: Changed.

File: Mesh/Mesh.cpp (hunk 4 above)
Comment (wenqing) [inline Mesh/Mesh.cpp]: Since this tool is only for ogs5, c++11 feature is not considered.

**Comment by wenqing** [comment]
@TomFischer  
The tool was programmed for ogs5, and no programming style guide was applied at that time. There is 'using namespace std' in .cpp file, and it is removed in the last commit, a6fd925

File: Mesh/Mesh.cpp (hunk 3 above)
Comment (wenqing) [inline Mesh/Mesh.cpp]: removed

File: Mesh/Mesh.cpp [hunk 5]
```diff
@@ -891,14 +891,11 @@ void Mesh::ConstructSubDomain_by_Nodes(const MeshPartConfig mpc)
    const MyInt nn = static_cast<MyInt>(node_vector.size());
 
    vector<bool> sdom_marked(nn);
-   vector<MyInt> dom_idx(NodesNumber_Linear);
+   vector<MyInt> dom_idx(nn);
```
Comment (wenqing) [inline Mesh/Mesh.cpp]: removed.

**Comment by TomFischer** [comment]
Sorry, I did not know that the tool is for OGS5.

**Comment by wenqing** [comment]
@TomFischer If you have any further comments, you can merge it. It's for OGS5.

**Comment by endJunction** [comment]
@wenqing Neither Tom nor me can merge this, because we don't have an overview what/who will be affected. Feel free to merge it yourself when you a sure about the changes.

**Pull request merged**